### ISSUE TITLE
Stale .travis.yml script is used for build after rebase onto master

### ISSUE THREAD

**Issue by calcmogul** [travis-ci/travis-ci#8796]
When building PR 38 in wpilibsuite/allwpilib, Travis uses a .travis.yml much older than the one in the pull request (even after rebasing upon master and clearing the Travis cache entry for that PR). See https://travis-ci.org/wpilibsuite/allwpilib/builds/303708231.

We removed the .travis.yml on August 18, 2017 in commit e1195e8b, then added a different config back on August 27, 2017 in commit 67d4da51. The config Travis uses is older than the one removed on August 18.

My theory is that Travis stopped pulling an updated .travis.yml for that PR build once the config got deleted from master (even though it got added back later). The config Travis is using is probably older because I hadn't rebased that PR onto master before we removed the config.

**Comment by BanzaiMan** [comment]
As explained in https://docs.travis-ci.com/user/pull-requests/, we build the merge commit of the PR. If the PR is not up to date with the base branch, then you won't get the recent changes which are only reflected in the base branch.

**Comment by calcmogul** [comment]
You are correct that it's merging against an old commit on master (one from May 7th apparently). Why is it doing that when I already rebased the PR branch onto the tip of master? Does that not do what I think it does?

**Comment by calcmogul** [comment]
Looking at this issue a bit more, it looks like wpilibsuite/allwpilib#38 is exhibiting the same problem as #8795. https://travis-ci.org/wpilibsuite/allwpilib/builds/303708231#L442 shows that it's running `git fetch origin +refs/pull/384/merge` instead of what it should be, `git fetch origin +refs/pull/38/merge`. That would explain why my rebases aren't doing anything.
